applications/StructuralMechanicsApplication/custom_constitutive/timoshenko_beam_elastic_constitutive_law.cpp

```diff
@@ -0,0 +1,146 @@
+    const Properties& rMaterialProperties,
+    const GeometryType& rElementGeometry,
+    const ProcessInfo& rCurrentProcessInfo
+) const
+{
+    KRATOS_ERROR_IF_NOT(rMaterialProperties.Has(YOUNG_MODULUS)) << "YOUNG_MODULUS is not defined in the properties" << std::endl;
+    KRATOS_ERROR_IF_NOT(rMaterialProperties.Has(POISSON_RATIO)) << "POISSON_RATIO is not defined in the properties" << std::endl;
+    KRATOS_ERROR_IF_NOT(rMaterialProperties.Has(CROSS_AREA))    << "CROSS_AREA is not defined in the properties"    << std::endl;
+    KRATOS_ERROR_IF_NOT(rMaterialProperties.Has(DENSITY))       << "DENSITY is not defined in the properties"       << std::endl;
+    KRATOS_ERROR_IF_NOT(rMaterialProperties.Has(IZ))            << "IZ is not defined in the properties"            << std::endl;
+    return 0;
+}
+
+//************************************************************************************
+//************************************************************************************
+
+void TimoshenkoBeamElasticConstitutiveLaw::CalculateMaterialResponsePK2(ConstitutiveLaw::Parameters& rValues)
+{
+    CalculateMaterialResponseCauchy(rValues);
+}
+
+/***********************************************************************************/
+/***********************************************************************************/
+
+void TimoshenkoBeamElasticConstitutiveLaw::CalculateMaterialResponseKirchhoff(ConstitutiveLaw::Parameters& rValues)
+{
+    CalculateMaterialResponseCauchy(rValues);
+}
+
+/***********************************************************************************/
+/***********************************************************************************/
+
+void TimoshenkoBeamElasticConstitutiveLaw::CalculateMaterialResponsePK1(ConstitutiveLaw::Parameters& rValues)
+{
+    CalculateMaterialResponseCauchy(rValues);
+}
+
+/***********************************************************************************/
+/***********************************************************************************/
+
+void TimoshenkoBeamElasticConstitutiveLaw::CalculateMaterialResponseCauchy(ConstitutiveLaw::Parameters& rValues)
+{
+    const auto& r_cl_law_options = rValues.GetOptions();
+    auto &r_material_properties = rValues.GetMaterialProperties();
+    auto &r_strain_vector = rValues.GetStrainVector();
+    auto strain_size = GetStrainSize();
+
+    const double axial_strain = r_strain_vector[0]; // E_l
+    const double curvature    = r_strain_vector[1]; // Kappa
+    const double shear_strain = r_strain_vector[2]; // Gamma_xy
+
+    const double E    = r_material_properties[YOUNG_MODULUS];
+    const double nu   = r_material_properties[POISSON_RATIO];
+    const double A    = r_material_properties[CROSS_AREA];
+    const double I    = r_material_properties[IZ];
+    const double k_s  = r_material_properties.Has(SHEAR_CORRECTION_XY) ? r_material_properties[SHEAR_CORRECTION_XY] : 5.0 / 6.0; // We assume rectangular shape
```

Comment:
I guess this means that it is designed for only 2D

Reply:
Yes, it is.